### PR TITLE
Added Face emotion training and testing files

### LINKED ISSUE #988: 🧑‍💻: Face Emotion Recognition
### Title

In this project the face expressions will be classified into 7 types of expression that is sad, happy ,surprise, angry ,neutral , disgust,  

### Enhancement Aim

To implement a comprehensive face emotion detection and classification system that works with both live camera feed and uploaded video files, enhancing the application's capability to analyze and interpret human emotions in real-time.

### Changes

Changes

Core Functionality

Implement face detection module for real-time camera feed
Add emotion classification model integration
Create video file processing pipeline
Develop multi-face tracking system

Technical Implementation

Camera integration for live feed capture
Video file upload and processing functionality
Real-time emotion classification processing
UI components for visualization and controls
Backend API endpoints for video processing

User Interface

Live camera preview window
Video upload interface
Emotion classification results display
Control panel for camera/video switching
Processing status indicators

### Screenshots 📷

_No response_

### Guidelines

- [X] I have read the guidelines
- [X] I have the link to my latest merged PR

### Full Name

shiven yadav s

### Participant Role

GSSOC, HACKTOBERFEST

### PR DESCRIPTION
**Pull request by shivenyadavs**
### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
**Please enter the title of the issue related to your pull request.**  
*Face Emotion Recognition*
closes #988 

- [X] I have provided the issue title.

---

#### Info about the Related Issue
**What's the goal of the project?** 
The goal of this project is to develop a comprehensive face emotion detection and classification system. This system will classify facial expressions into seven categories (sad, happy, surprise, angry, neutral, disgust) and work with both live camera feed and uploaded video files. This aims to enhance the application's ability to analyze and interpret human emotions in real-time. 
*Describe the aim of the project.*
The aim of the project is to create a robust face emotion detection and classification system that can identify seven key facial expressions: sad, happy, surprise, angry, neutral, disgust, and fear. This system is designed to work with both live camera feeds and uploaded video files, providing a real-time analysis of human emotions. By integrating face detection and multi-face tracking capabilities, the project aims to offer accurate and dynamic emotion interpretation, which could have applications in user experience research, mental health monitoring, and interactive systems.
- [X] I have described the aim of the project.

---

#### Name
**Please mention your name.**  
*Shiven Yadav S*

- [X] I have provided my name.

---

#### GitHub ID
**Please mention your GitHub ID.**  
*https://github.com/shivenyadavs*

- [X] I have provided my GitHub ID.

---

#### Email ID
**Please mention your email ID for further communication.**  
*shivenyadavs@gmail.com*

- [X] I have provided my email ID.

---

#### Identify Yourself
**Mention in which program you are contributing (e.g., WoB, GSSOC, SSOC, SWOC).**  
*GSSOC*

- [X] I have mentioned my participant role.

---

#### Closes
**Enter the issue number that will be closed through this PR.**  
*Closes: #988 *

- [X] I have provided the issue number.

---

#### Describe the Add-ons or Changes You've Made
**Give a clear description of what you have added or modified.**  
*Describe your changes here.*
Model Training (train.py):

Implemented a script to train the facial emotion recognition model using a dataset with labeled images for various emotions. The model learns to classify expressions by analyzing the features in the training images.
Added data preprocessing steps, such as resizing images and normalizing pixel values, to prepare the dataset for training.
Configured model architecture, training parameters (like batch size and epochs), and an optimizer to maximize accuracy.
Face Detection and Emotion Classification (test.py):

Integrated face detection using haarcascade.xml in test.py, enabling real-time detection of faces through a camera feed.
Loaded the trained model to classify detected faces into emotion categories based on the trained model’s predictions.
Developed a display feature that shows the recognized expression label (e.g., "happy" or "sad") directly on the screen above the detected face.
Dataset Utilization:

Set up a structured train and test dataset with labeled images for different facial expressions to support model training and testing.
- [X] I have described my changes.

---

#### Type of Change
**Select the type of change:**  
- [] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [] Code style update (formatting, local variables)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

---

#### How Has This Been Tested?
**Describe how your changes have been tested.**  
*Model Training Validation:

After training the model in train.py, evaluated its performance on the test dataset by calculating accuracy and checking confusion matrices to ensure the model accurately classifies each emotion.
Adjusted parameters as necessary to optimize accuracy, then retrained and re-evaluated.
Real-Time Emotion Detection Test (test.py):

Ran test.py to test real-time face and emotion detection using a live camera feed.
Tested the system with various facial expressions to verify that the correct emotion label appears on the screen.
Adjusted face detection and classification parameters to ensure smooth and accurate detection in different lighting conditions and with multiple faces.
Edge Cases:

Tested edge cases, such as partial face visibility and different angles, to ensure consistent and reliable emotion recognition.
Verified the model’s stability by running extended tests with continuous camera feed to check for performance issues or crashes..*

- [X] I have described my testing process.

---

#### Checklist
**Please confirm the following:**  
- [X] My code follows the guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly wherever it was hard to understand.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] I have added things that prove my fix is effective or that my feature works.
- [X] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by shivenyadavs** [comment]
@UTSAVS26 please check the changes.

**Comment by shivenyadavs** [comment]
@UTSAVS26 if there are any changes please let me know

**Comment by UTSAVS26** [comment]
@shivenyadavs why the code is commented?

**Comment by shivenyadavs** [comment]
those are alternate codes which will work.

**Comment by shivenyadavs** [comment]
The main working code is not commented please check

**Comment by UTSAVS26** [comment]
> those are alternate codes which will work.

So create a separate file and add a readme file for telling the different ways to do the same work.

**Comment by shivenyadavs** [comment]
I have made the changes in emotion detection readme.md please have a look

File: EmotionClassification/Readme.md
Comment (UTSAVS26) [inline EmotionClassification/Readme.md:1]: Do proper formatting.

File: EmotionClassification/Testemotion.py
Comment (UTSAVS26) [inline EmotionClassification/Testemotion.py:1]: Make two separate files for it, don't just comment the whole section of the code.

**Comment by UTSAVS26** [comment]
@shivenyadavs don't add the project in the main folder, move it in some dedicated folders.

**Comment by shivenyadavs** [comment]
done please check

**Comment by UTSAVS26** [comment]
@shivenyadavs add the project in dedicated folders not into the main repo.

File: EmotionClassification/emotion-detection-readme.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+   - Check if your webcam is properly connected
+   - Try changing the camera index (e.g., `cv2.VideoCapture(1)`)
+
+2. **Low detection accuracy**
+   - Ensure good lighting conditions
+   - Keep face centered and at a reasonable distance
+   - Check if the face is clearly visible and not obscured
+
+3. **Performance issues**
+   - Reduce frame resolution
```
Comment (ruhi47) [inline EmotionClassification/emotion-detection-readme.md:155]: Please work on README files because it is creating lots of confusion and is not relevant on some steps.

File: EmotionClassification/Readme.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+# Facial Emotion Detection with OpenCV and Keras
```
Comment (ruhi47) [inline EmotionClassification/Readme.md:1]: Hey @shivenyadavs ,
First of all, do not push from `main` branch to 'main'.
Create a 'feature', 'develop' or 'hotfix' feature branch from  your fork and then push it.

I suggest you go through my comments below, do the changes and then push again a new PR and tag me.

Do the required changes and I will go through your PR again. 

Thank you and all the best!

**Review by ruhi47 — CHANGES_REQUESTED** [review]
